code/modules/mob/living/silicon/ai/freelook/chunk.dm

```diff
@@ -1,105 +1,98 @@
 	var/list/seenby = list()
-	var/changed = 0
+	var/changed = FALSE
 	var/x = 0
 	var/y = 0
 	var/z = 0
 
-// Add an AI eye to the chunk, then update if changed.
-
+/// Add an AI eye to the chunk, then update if changed.
 /datum/camerachunk/proc/add(mob/camera/ai_eye/eye)
 	eye.visibleCameraChunks += src
 	seenby += eye
 	if(changed)
 		update()
 
-// Remove an AI eye from the chunk, then update if changed.
-
+/// Remove an AI eye from the chunk, then update if changed.
 /datum/camerachunk/proc/remove(mob/camera/ai_eye/eye, remove_static_with_last_chunk = TRUE)
 	eye.visibleCameraChunks -= src
 	seenby -= eye
+
 	if(remove_static_with_last_chunk && !eye.visibleCameraChunks.len)
 		var/client/client = eye.GetViewerClient()
-		if(client)
-			switch(eye.use_static)
-				if(USE_STATIC_TRANSPARENT)
-					client.images -= GLOB.cameranet.obscured_transparent
-				if(USE_STATIC_OPAQUE)
-					client.images -= GLOB.cameranet.obscured
-
-// Called when a chunk has changed. I.E: A wall was deleted.
+		if(client && eye.use_static)
+			client.images -= GLOB.cameranet.obscured
 
+/// Called when a chunk has changed. I.E: A wall was deleted.
 /datum/camerachunk/proc/visibilityChanged(turf/loc)
 	if(!visibleTurfs[loc])
 		return
 	hasChanged()
 
-// Updates the chunk, makes sure that it doesn't update too much. If the chunk isn't being watched it will
-// instead be flagged to update the next time an AI Eye moves near it.
-
+/**
+ * Updates the chunk, makes sure that it doesn't update too much. If the chunk isn't being watched it will
+ * instead be flagged to update the next time an AI Eye moves near it.
+ */
 /datum/camerachunk/proc/hasChanged(update_now = 0)
 	if(seenby.len || update_now)
-		addtimer(CALLBACK(src, .proc/update), UPDATE_BUFFER, TIMER_UNIQUE)
+		addtimer(CALLBACK(src, .proc/update), UPDATE_BUFFER_TIME, TIMER_UNIQUE)
 	else
-		changed = 1
-
-// The actual updating. It gathers the visible turfs from cameras and puts them into the appropiate lists.
+		changed = TRUE
 
+/// The actual updating. It gathers the visible turfs from cameras and puts them into the appropiate lists.
 /datum/camerachunk/proc/update()
 	var/list/newVisibleTurfs = list()
 
-	for(var/camera in cameras)
-		var/obj/machinery/camera/c = camera
-
-		if(!c)
+	for(var/obj/machinery/camera/current_camera as anything in cameras)
+		if(!current_camera)
 			continue
 
-		if(!c.can_use())
+		if(!current_camera.can_use())
 			continue
 
 		var/turf/point = locate(src.x + (CHUNK_SIZE / 2), src.y + (CHUNK_SIZE / 2), src.z)
-		if(get_dist(point, c) > CHUNK_SIZE + (CHUNK_SIZE / 2))
+		if(get_dist(point, current_camera) > CHUNK_SIZE + (CHUNK_SIZE / 2))
 			continue
 
-		for(var/turf/t in c.can_see())
-			// Possible optimization: if(turfs[t]) here, rather than &= turfs afterwards.
-			// List associations use a tree or hashmap of some sort (alongside the list itself)
-			//  so are surprisingly fast. (significantly faster than var/thingy/x in list, in testing)
```

Comment:
Did this get tested? I assume it scales poorly depending on the amount of turfs in can_see, tho maybe does &= in n^2 time or something

Reply:
no idea if it ever got tested, i can look at it tomorrow

Reply:
split it off into a subproc that calls the two based on prob, and see which performs the worst when you're doing a lot of "this"

Reply:
![image](https://user-images.githubusercontent.com/58055496/118609921-1b77fa80-b770-11eb-9e2c-5a779cfce99e.png)
Let's jive

Reply:
as far as i can tell the new way is reasonably faster